### PR TITLE
fix: re-add name and description editing on tool mode except for Composio, fix MCP server code

### PR DESCRIPTION
**Pull request by lucaseduoli**
This pull request introduces a new `readonly` property for tools in the system, ensuring that tools marked as `readonly` are handled appropriately in both the backend and frontend. The changes span metadata configuration, metadata input building, and frontend behavior adjustments.

### Backend Changes:
* **Tool Metadata Configuration (`src/backend/base/langflow/base/composio/composio_base.py`)**: Added a `readonly` property to the `metadata` dictionary of tools, defaulting to `True` during tool configuration.
* **Metadata Input Building (`src/backend/base/langflow/custom/custom_component/component.py`)**: Updated the `_build_tools_metadata_input` method to include the `readonly` property from tool metadata, defaulting to `False` if not set.

### Frontend Changes:
* **Tools Table Behavior (`src/frontend/src/modals/toolsModal/components/toolsTable/index.tsx`)**: Modified the `ToolsTable` component to respect the `readonly` property, preventing actions on tools marked as `readonly`.

### CODE REVIEW

**Comment by edwinjosechittilappilly** [comment]
sounds good to me. 
Will test and approve it.
Good Work @lucaseduoli

File: src/backend/base/langflow/base/tools/component_tool.py [hunk 1]
```diff
@@ -40,21 +39,8 @@ def _get_input_type(input_: InputTypes):
...
-                [
-                    f"{name}. {input_name}: {_get_input_type(component._inputs[input_name])}"
-                    for input_name in output.required_inputs
-                ]
-            )
-        )
-    else:
-        args = ""
-    return f"{name}. {output.method}({args}) - {component.description}"
+    return f"{name}. {output.method} - {component.description}"
```
Comment (ogabrielluiz) [inline src/backend/base/langflow/base/tools/component_tool.py:43]: We must pass the args because it helps the LLM understand how to use the tool

**Review by ogabrielluiz — CHANGES_REQUESTED** [review]

File: src/backend/base/langflow/base/tools/component_tool.py (hunk 1 above)
Comment (edwinjosechittilappilly) [reply src/backend/base/langflow/base/tools/component_tool.py:43]: @ogabrielluiz  Doent the arg schema already have that data in StructuredTool?

**Comment by edwinjosechittilappilly** [comment]
@ogabrielluiz  Also there seems to be a Bug when testing with Tavily search component the args seems to be empty.

File: src/backend/base/langflow/base/tools/component_tool.py (hunk 1 above)
Comment (ogabrielluiz) [reply src/backend/base/langflow/base/tools/component_tool.py:43]: It should. I'm not sure it is supported as well in all providers as it is on OpenAI. I'll trust that this not necessary, then.

**Review by ogabrielluiz — APPROVED** [review]

**Review by edwinjosechittilappilly — APPROVED** [review]
LGTM
